src/main.rs

```diff
@@ -247,6 +248,27 @@ impl SendState {
     }
 }
 
+fn all_filenames_from_path_names(fnames: &[String]) -> Vec<String> {
+    let mut all_files = Vec::with_capacity(fnames.len());
+
+    for fname in fnames {
+        let metadata = std::fs::metadata(fname).unwrap();
+
+        if metadata.is_file() {
+            all_files.push(fname.to_owned());
+        } else if metadata.is_dir() {
+            all_files.extend(
+                WalkDir::new(fname)
+                    .into_iter()
+                    .filter_map(|x| x.ok())
+                    .filter(|e| e.file_type().is_file())
+                    .filter_map(|e| e.path().to_str().map(String::from)),
```

Comment:
If it becomes annoying to convert between `Path` and `String` all the time, we could also change the function to return `PathBuf`, and send that in the transfer plan. But then printing them becomes a bit messier … I think it’s okay to stick with `String`, we don’t _fully_ control all the files we need to transfer but if something we run is producing non-UTF-8 file names then we should go yell at the authors.